### PR TITLE
Add Enterprise Edition page to Architecture menu

### LINKED ISSUE #1319: KKP EE vs CC feature comparison
Create a page in the docs where EE and CE version of KKP are compared to each other among their differences in features.

### PR DESCRIPTION
**Pull request by csengerszabo**
Adding a new page under Architecture menu that describes the differences and extra features of Enterprise Edition.

Fixes #1319

### CODE REVIEW

**Comment by embik** [comment]
So this is just my personal two cents, but I wonder if we should do a "feature table" that lists both CE and EE features and presents checkmarks for the respective feature being supported in CE and EE or EE only?

**Comment by csengerszabo** [comment]
@embik, yes, that is the long-term vision. But on the road from not having any kind of EE vs CE page to the end goal of this full feature matrix, this is a tangible step in the middle with useful information for the public imho. For that next step, we would have to collect and list all the features of CE which will be a longer task.

File: content/kubermatic/main/architecture/enterprise-edition/_index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
++++
+title = "Enterprise edition"
```
Comment (embik) [inline content/kubermatic/main/architecture/enterprise-edition/_index.md:3]:
```suggestion
title = "Enterprise Edition"
```

File: content/kubermatic/main/architecture/enterprise-edition/_index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+date = 2023-01-31T12:11:35+02:00
+weight = 6
+
++++
+
+## Why is there Enterprise Edition and Community Edition
+
+Kubermatic Kubernetes Platform (KKP) is an open-source product, and both the Community Edition and the Enterprise Edition are available to download from GitHub. However, there are some extra features that are only available in the Enterprise Edition of Kubermatic Kubernetes Platform.
+
+## Extra features of Enterprise Edition
```
Comment (embik) [inline content/kubermatic/main/architecture/enterprise-edition/_index.md:13]:
```suggestion
## Extra Features of Enterprise Edition
```

**Review by embik — CHANGES_REQUESTED** [review]
@csengerszabo gotcha, makes sense. Then I just have two nitpicks about title casing for the headers.

**Comment by embik** [comment]
/hold

quick clarification internally.

**Pull request merged**